packages/dapp/src/components/listing/ChallengeDetail.tsx

```diff
@@ -197,15 +216,25 @@ const mapStateToProps = (
     if (challengeUserDataMap) {
       userChallengeData = challengeUserDataMap.get(userAcct.account);
     }
+    if (challengeData) {
+      const wrappedChallenge = challengeData as WrappedChallengeData;
+      if (wrappedChallenge && wrappedChallenge.challenge && wrappedChallenge.challenge.appeal) {
```

Comment:
same, here this big inline boolean statements make everything a lot harder to follow